velox/dwio/common/SelectiveRepeatedColumnReader.h

```diff
@@ -54,102 +53,9 @@ class SelectiveRepeatedColumnReader : public SelectiveColumnReader {
       int32_t numLengths,
       const uint64_t* FOLLY_NULLABLE nulls) = 0;
 
-  void makeNestedRowSet(RowSet rows) {
-    allLengths_.resize(rows.back() + 1);
-    assert(!allLengths_.empty()); // for lint only.
-    auto nulls =
-        nullsInReadRange_ ? nullsInReadRange_->as<uint64_t>() : nullptr;
-    // Reads the lengths, leaves an uninitialized gap for a null
-    // map/list. Reading these checks the null nask.
-    readLengths(allLengths_.data(), rows.back() + 1, nulls);
-    dwio::common::ensureCapacity<vector_size_t>(
-        offsets_, rows.size(), &memoryPool_);
-    dwio::common::ensureCapacity<vector_size_t>(
-        sizes_, rows.size(), &memoryPool_);
-    auto rawOffsets = offsets_->asMutable<vector_size_t>();
-    auto rawSizes = sizes_->asMutable<vector_size_t>();
-    vector_size_t nestedLength = 0;
-    for (auto row : rows) {
-      if (!nulls || !bits::isBitNull(nulls, row)) {
-        nestedLength += allLengths_[row];
-      }
-    }
-    nestedRows_.resize(nestedLength);
-    vector_size_t currentRow = 0;
-    vector_size_t nestedRow = 0;
-    vector_size_t nestedOffset = 0;
-    for (auto rowIndex = 0; rowIndex < rows.size(); ++rowIndex) {
-      auto row = rows[rowIndex];
-      // Add up the lengths of non-null rows skipped since the last
-      // non-null.
-      for (auto i = currentRow; i < row; ++i) {
-        if (!nulls || !bits::isBitNull(nulls, i)) {
-          nestedOffset += allLengths_[i];
-        }
-      }
-      currentRow = row + 1;
-      // Check if parent is null after adding up the lengths leading
-      // up to this. If null, add a null to the result and keep
-      // looping. If the null is last, the lengths will all have been
-      // added up.
-      if (nulls && bits::isBitNull(nulls, row)) {
-        rawOffsets[rowIndex] = 0;
-        rawSizes[rowIndex] = 0;
-        bits::setNull(rawResultNulls_, rowIndex);
-        anyNulls_ = true;
-        continue;
-      }
-
-      auto lengthAtRow = allLengths_[row];
-      std::iota(
-          &nestedRows_[nestedRow],
-          &nestedRows_[nestedRow + lengthAtRow],
-          nestedOffset);
-      rawOffsets[rowIndex] = nestedRow;
-      rawSizes[rowIndex] = lengthAtRow;
-      nestedRow += lengthAtRow;
-      nestedOffset += lengthAtRow;
-    }
-    childTargetReadOffset_ += nestedOffset;
-  }
+  void makeNestedRowSet(RowSet);
```

Comment:
Thank you for moving these functions to the .cpp file. Would you add documentation and parameter name?